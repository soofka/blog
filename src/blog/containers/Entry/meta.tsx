import { inject, observer } from 'mobx-react';
import * as React from 'react';

import { getLabel } from 'blog/common/helpers';
import { LanguageStoreInterface } from 'blog/store/language';

import { Meta } from 'blog/components/meta';

interface EntryMetaPropsInterface {
  title: string;
  description: string;
  keywords: string[];
  imageUrl: string;
  languageStore?: LanguageStoreInterface;
}

export const EntryMetaComponent = (props: EntryMetaPropsInterface) => {
  const { title, description, keywords, imageUrl, languageStore: { getLanguage } } = props;
  return <Meta
    title={getLabel('page_title__entry', { entry_title: title }, getLanguage())}
    description={getLabel('page_description', { description }, getLanguage())}
    keywords={getLabel('page_keywords', { keywords: keywords.join(', ') }, getLanguage())}
    imageUrl={imageUrl}
  />;
};

export const EntryMeta = inject('languageStore')(observer(EntryMetaComponent));
