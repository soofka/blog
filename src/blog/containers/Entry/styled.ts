import styledComponents from 'styled-components';

export const StyledEntry = styledComponents.div`
  margin: ${(props) => props.theme.marginM} 0;
  padding: ${(props) => props.theme.paddingM};
  box-sizing: border-box;

  &:first-of-type {
    margin-top: 0;
  }

  &:last-of-type {
    margin-bottom: 0;
  }

  @media (max-width: ${(props) => props.theme.breakpointM}) {
    margin: ${(props) => props.theme.marginS} 0;
    padding: ${(props) => props.theme.paddingS};

    blockquote {
      margin-left: 0;
      margin-right: 0;
    }
  }
`;

export const StyledEntryContent = styledComponents.div`
  small {
    display: block;
    width: 100%;
    font-size: ${(props) => props.theme.fontSizeS};
    line-height: ${(props) => props.theme.lineHeightS};
    margin: ${(props) => props.theme.marginM} 0;
    padding: 0;
    color: ${(props) => props.theme.gray};

    a {
      color: ${(props) => props.theme.gray};
      text-decoration: underline;

      &:hover,
      &:active {
        color: ${(props) => props.theme.gray};
        text-decoration: underline;
      }
    }
  }

  p,
  li {
    font-size: ${(props) => props.theme.fontSizeM};
    line-height: ${(props) => props.theme.lineHeightL};
    margin: 0 0 ${(props) => props.theme.marginL} 0;
    padding: 0;
    text-align: justify;

    /* stylelint-disable-next-line */
    a {
      color: ${(props) => props.theme.blue};

      &:hover,
      &:active {
        color: ${(props) => props.theme.lighterBlue};
      }
    }
  }

  li {
    margin: 0;
  }

  blockquote {
    color: ${(props) => props.theme.blue};
    margin: 0 ${(props) => `${props.theme.marginXL} ${props.theme.marginL} ${props.theme.marginXL}`};
    padding: 0 ${(props) => props.theme.paddingM};
    font-size: ${(props) => props.theme.fontSizeXXXL};
    line-height: ${(props) => props.theme.fontSizeXXXXL};
    border-left: 4px solid ${(props) => props.theme.lighterBlue};
  }

  img {
    display: block;
    max-width: 100%;
    margin-left: auto;
    margin-right: auto;
    padding: 0;
  }

  ul {
    margin: 0;
    padding: 0 0 0 ${(props) => props.theme.paddingL};
  }

  pre code {
    background-color: ${(props) => props.theme.blue};
  }

  hr {
    width: 70%;
    height: 1px;
    margin: ${(props) => props.theme.marginXXL} auto;
    border: 0;
    border-top: 1px solid ${(props) => props.theme.darkerWhite};
  }

  .video-container {
    max-width: 800px;
    max-height: 450px;
    margin: ${(props) => `${props.theme.marginS} auto ${props.theme.marginM} auto`};
    overflow: hidden;
  }

  .video-container--wrapper {
    position: relative;
    padding-top: 24px;
    padding-bottom: 56.25%;
    height: 0;

    iframe {
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;
      height: 100%;
    }
  }
`;
