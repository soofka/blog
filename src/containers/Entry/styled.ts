import styledComponents from 'styled-components';

export const StyledEntry = styledComponents.div`
  margin: ${props => props.theme.marginM} 0;
  padding: ${props => props.theme.paddingM};
  box-sizing: border-box;

  &:first-of-type {
    margin-top: 0;
  }

  &:last-of-type {
    margin-bottom: 0;
  }

  @media (max-width: ${props => props.theme.mobileBreakPointWidth}) {
    margin: ${props => props.theme.marginS} 0;
    padding: ${props => props.theme.paddingS};

    blockquote {
      margin-left: 0;
      margin-right: 0;
    }
  }
`;

export const StyledEntryContent = styledComponents.div`
  small {
    display: block;
    width: 100%;
    font-size: ${props => props.theme.fontSizeS};
    line-height: ${props => props.theme.fontSizeL};
    margin: 0 0 ${props => props.theme.marginM} 0;
    padding: 0;
    color: ${props => props.theme.gray};

    a {
      color: ${props => props.theme.gray};
      text-decoration: underline;

      &:hover,
      &:active {
        color: ${props => props.theme.gray};
        text-decoration: underline;
      }
    }
  }

  p,
  li {
    font-size: ${props => props.theme.fontSizeL};
    line-height: ${props => props.theme.fontSizeXXL};
    margin: 0 0 ${props => props.theme.marginL} 0;
    padding: 0;
    text-align: justify;

    /* stylelint-disable-next-line */
    a {
      color: ${props => props.theme.blue};

      &:hover,
      &:active {
        color: ${props => props.theme.lighterBlue};
      }
    }
  }

  li {
    margin: 0;
  }

  pre {
    margin: 0;
    padding: 0;
  }

  code {
    float: left;
    width: 100%;
    margin: 0 0 ${props => props.theme.marginL} 0;
    padding: ${props => props.theme.paddingM};
    color: ${props => props.theme.black};
    background-color: ${props => props.theme.darkWhite};
    box-sizing: border-box;
  }

  blockquote {
    color: ${props => props.theme.blue};
    margin: 0 ${props => `${props.theme.marginXL} ${props.theme.marginL} ${props.theme.marginXL}`};
    padding: 0 ${props => props.theme.paddingM};
    font-size: ${props => props.theme.fontSizeXXXL};
    border-left: 4px solid ${props => props.theme.lighterBlue};
    line-height: ${props => props.theme.fontSizeXXXXL};
  }

  img {
    display: block;
    max-width: 100%;
    margin: ${props => props.theme.marginS} auto;
    padding: 0;
    border: 2px solid ${props => props.theme.white};
    outline: 1px solid ${props => props.theme.lighterBlack};
  }

  ul {
    margin: 0;
    padding: 0 0 0 ${props => props.theme.paddingL};
  }

  hr {
    margin: ${props => props.theme.marginXXL};
  }

  .video-container {
    max-width: 800px;
    max-height: 450px;
    margin: ${props => `${props.theme.marginS} auto ${props.theme.marginM} auto`};
    overflow: hidden;
  }

  .video-container--wrapper {
    position: relative;
    padding-top: 24px;
    padding-bottom: 56.25%;
    height: 0;

    iframe {
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;
      height: 100%;
    }
  }
`;
