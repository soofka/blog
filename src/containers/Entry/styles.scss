@import 'components/Blog/variables.scss';

.entry {
  margin: $marginM 0;
  padding: $paddingM;
  box-sizing: border-box;

  &:first-of-type {
    margin-top: 0;
  }

  &:last-of-type {
    margin-bottom: 0;
  }

  &-content {
    small {
      display: block;
      width: 100%;
      font-size: $fontSizeS;
      line-height: $fontSizeL;
      margin: 0 0 $marginM 0;
      padding: 0;
      color: $gray;

      a {
        color: $gray;
        text-decoration: underline;

        &:hover,
        &:active {
          color: $gray;
          text-decoration: underline;
        }
      }
    }

    p,
    li {
      font-size: $fontSizeL;
      line-height: $fontSizeXXL;
      margin: 0 0 $marginL 0;
      padding: 0;
      text-align: justify;

      /* stylelint-disable */
      a {
      /* stylelint-enable */
        color: $blue;

        &:hover,
        &:active {
          color: $lighterBlue;
        }
      }
    }

    li {
      margin: 0;
    }

    code {
      float: left;
      width: 100%;
      margin: 0 0 $marginL 0;
      padding: $paddingM;
      color: $black;
      background-color: $darkWhite;
      box-sizing: border-box;

      pre {
        margin: 0;
        padding: 0;
      }
    }

    blockquote {
      color: $blue;
      margin: 0 $marginXL $marginL $marginXL;
      padding: 0 $paddingM;
      font-size: $fontSizeXXXL;
      border-left: 4px solid $lighterBlue;
      line-height: $fontSizeXXXXL;
    }

    img {
      display: block;
      max-width: 100%;
      margin: $marginS auto;
      padding: 0;
      border: 2px solid $white;
      outline: 1px solid $lighterBlack;
    }

    ul {
      margin: 0;
      padding: 0 0 0 $paddingL;
    }

    hr {
      margin: $marginXXL;
    }

    .video-container {
      max-width: 800px;
      max-height: 450px;
      margin: $marginS auto $marginM auto;
      overflow: hidden;

      &--wrapper {
        position: relative;
        padding-top: 24px;
        padding-bottom: 56.25%;
        height: 0;

        iframe {
          position: absolute;
          top: 0;
          left: 0;
          width: 100%;
          height: 100%;
        }
      }
    }
  }
}

@media (max-width: $mobileBreakPointWidth) {
  .entry {
    margin: $marginS 0;
    padding: $paddingS;

    blockquote {
      margin-left: 0;
      margin-right: 0;
    }
  }
}
