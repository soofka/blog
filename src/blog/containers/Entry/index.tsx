import { reaction } from 'mobx';
import { inject } from 'mobx-react';
import * as React from 'react';

import {
  getEntryContent,
  RequestSuccessResponseInterface,
  scrollToEntryTop,
  validateRequestSuccessResponse,
} from 'blog/common/helpers';
import { EntryBrief } from 'blog/components/EntryBrief';
import { EntryContent } from 'blog/components/EntryContent';
import { EntryFooter } from 'blog/components/EntryFooter';
import { EntryHeader } from 'blog/components/EntryHeader';
import { ErrorBox } from 'blog/components/ErrorBox';
import { Label } from 'blog/components/Label';
import { LoadingCover } from 'blog/components/LoadingCover';
import { LanguageStoreInterface } from 'blog/store/language';
import { getEntryFilePath } from 'common/helpers';
import { EntryInterface, TagInterface } from 'common/types';
import { EntryMeta } from './meta';

import { StyledEntry } from './styled';

export interface EntryPropsInterface extends EntryInterface, React.Props<any> {
  languageStore?: LanguageStoreInterface;
}

interface EntryStateInterface {
  content: string;
  loading: boolean;
}

export class EntryContainer extends React.Component<EntryPropsInterface, EntryStateInterface> {
  constructor(props: EntryPropsInterface) {
    super(props);
    this.state = { content: undefined, loading: false };

    reaction(
      () => this.props.languageStore.language,
      () => this.getContent(),
    );
  }

  componentDidMount() {
    if (!this.state.content) {
      this.getContent();
    }
  }

  getContent() {
    this.setState({ loading: true });
    const { contentFileName, languageStore: { getLanguage } } = this.props;

    getEntryContent(getEntryFilePath(contentFileName, getLanguage()))
      .then((response: RequestSuccessResponseInterface) => {
        const content = validateRequestSuccessResponse(response)
          ? response.data
          : undefined;

        this.setState({ content, loading: false });
        scrollToEntryTop();
      });
  }

  render() {
    const { title, tags, brief, niceUrl, imageUrl, created, updated } = this.props;
    const { content, loading } = this.state;

    return (
      <StyledEntry>
        <EntryMeta
          title={title}
          description={brief}
          keywords={tags.map((tag: TagInterface) => tag.name)}
          imageUrl={imageUrl}
        />
        <EntryHeader
          title={title}
          titleLink={niceUrl}
          tags={tags}
          created={created}
          updated={updated}
        />
        {loading &&
          <LoadingCover />
        }
        {!loading && !content &&
          <ErrorBox
            message={<Label name="errors__no_entry_content" />}
          />
        }
        {!loading && content &&
          <div>
            <EntryBrief
              text={brief}
              imageUrl={imageUrl}
            />
            <EntryContent
              content={content}
            />
            <EntryFooter
              title={title}
              niceUrl={niceUrl}
            />
          </div>
        }
      </StyledEntry>
    );
  }
}

export const Entry = inject('languageStore')(EntryContainer);
