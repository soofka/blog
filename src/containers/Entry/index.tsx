import * as React from 'react';
import { reaction } from 'mobx';
import { inject } from 'mobx-react';

import RequestHandler, { RequestHandlerResponseInterface } from 'common/RequestHandler';
import AssetsProvider from 'common/AssetsProvider';
import RoutingProvider from 'common/RoutingProvider';

import { LanguageStoreInterface } from 'store/language';

import Label from 'components/Label';
import EntryHeader from 'components/EntryHeader';
import EntryBrief, { EntryBriefInterface } from 'components/EntryBrief';
import LoadingCover from 'components/LoadingCover';
import ErrorBox from 'components/ErrorBox';
import EntryContent from 'components/EntryContent';
import EntryComments from 'components/EntryComments';
import ContentDivider from 'components/ContentDivider';

import { StyledEntry, StyledEntryContent } from './styled';

export interface EntryInterface {
  id: string;
  title: string;
  tags: string[];
  brief: EntryBriefInterface;
  created: string;
  updated: string;
  contentFileName: string;
  public?: boolean;
}

export interface EntryPropsInterface extends EntryInterface, React.Props<any> {
  full: boolean;
  languageStore?: LanguageStoreInterface;
}

interface EntryStateInterface {
  content: string;
  loading: boolean;
}

export class Entry extends React.Component<EntryPropsInterface, EntryStateInterface> {
  constructor(props: EntryPropsInterface) {
    super(props);
    this.state = { content: undefined, loading: false };

    reaction(
      () => this.props.languageStore.language,
      () => {
        if (this.props.full) {
          this.getContent();
        }
      },
    );
  }

  componentDidMount() {
    if (this.props.full && !this.state.content) {
      this.getContent();
    }
  }

  getContent() {
    this.setState({ loading: true });
    AssetsProvider
      .getEntryContent(AssetsProvider.getEntryFilePath(this.props.contentFileName))
      .then((response: RequestHandlerResponseInterface) => {

        const content = RequestHandler.validateResponse(response)
          ? response.data
          : undefined;
        this.setState({ content, loading: false });
      });
  }

  render() {
    const { title, tags, brief, created, updated, full } = this.props;
    const { content, loading } = this.state;

    const niceUrl = RoutingProvider.parseTextToNiceUrl(title);
    const niceUrlWithBasePath = `${RoutingProvider.getEntryNiceUrlBasePath()}/${niceUrl}`;

    return (
      <StyledEntry>
        <EntryHeader
          title={title}
          titleLink={niceUrlWithBasePath}
          tags={tags}
          created={created}
          updated={updated}
        />
        <StyledEntryContent>
          {!full &&
            <EntryBrief
              text={brief.text}
              full={false}
              imageFileName={brief.imageFileName}
              moreButtonLink={niceUrlWithBasePath}
            />}
          {full && loading &&
            <LoadingCover />}
          {full && !loading && !content &&
            <ErrorBox
              message={<Label name="errors__no_entry_content" />}
            />}
          {full && !loading && content &&
            <div>
              <EntryBrief
                text={brief.text}
                full={true}
                imageFileName={brief.imageFileName}
              />
              <EntryContent
                content={content}
              />
              <ContentDivider/>
              <EntryComments
                title={title}
                url={niceUrlWithBasePath}
                shortName={niceUrl}
                identifier={niceUrl}
              />
            </div>}
        </StyledEntryContent>
      </StyledEntry>
    );
  }
}

export default inject('languageStore')(Entry);
