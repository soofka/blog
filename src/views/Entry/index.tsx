import * as React from 'react';

import RoutingProvider from 'common/RoutingProvider';

import { Entries, EntriesParamsInterface, EntriesPropsInterface } from 'views/Entries';
import { EntryInterface } from 'containers/Entry';

export class Entry extends Entries {
  constructor(props: EntriesPropsInterface) {
    super(props);
    this.state = {
      entriesResponse: null,
      isEntryFull: true,
    };
  }

  filterEntries(entries: EntryInterface[], params: EntriesParamsInterface): EntryInterface[] {
    const { niceUrl } = params;

    /* tslint:disable-next-line */
    for (let entry of entries) {
      if (RoutingProvider.parseTextToNiceUrl(entry.title) === niceUrl) {
        return [entry];
      }
    }
    return [];
  }
}

export default Entry;
