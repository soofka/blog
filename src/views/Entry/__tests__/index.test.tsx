import * as React from 'react';
import { shallow } from 'enzyme';

import Entry from 'views/Entry';

describe('<Entry/>', () => {

  it('renders correctly', () => {
    expect(shallow(<Entry/>)).toMatchSnapshot();
  });

});
