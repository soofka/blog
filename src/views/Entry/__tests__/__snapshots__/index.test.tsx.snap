exports[`<Entry/> renders correctly 1`] = `<Component />`;
